src/builder-manifest.c

```diff
@@ -2827,13 +2832,27 @@ builder_manifest_cleanup (BuilderManifest *self,
               if (n_id)
                 {
                   n_text = n_id->first_child;
-                  if (n_text && g_strcmp0 (n_text->text, self->rename_desktop_file) == 0)
+                  if (n_text && g_strcmp0 (n_text->text, self->id) != 0)
                     {
-                      g_free (n_text->text);
+                      old_id = g_steal_pointer (&n_text->text);
                       n_text->text = g_strdup (self->id);
                     }
                 }
 
+              n_provides = flatpak_xml_find (n_root, "provides", NULL);
+              if (!n_provides)
+                {
+                  n_provides = flatpak_xml_new ("provides");
+                  flatpak_xml_add (n_root, n_provides);
+                }
+              if (n_provides)
```

Comment:
Will this always be true?

Reply:
likely. But I try to avoid null derefences.

Reply:
but then the same checks aren't done elsewhere.

Reply:
`flatpak_xml_new` can't fail and always returns something.